### PR TITLE
Fix TOB-4 nonces unique per token

### PR DESCRIPTION
**Pull request by wjuan-mob**
- Add filter in combine mint config and combine mint tx for duplicate nonces across different tokens
- Add tests for filtering duplicate nonces across different tokens
- Migrate LedgerDB to store nonces per token

### Motivation

This is part of a PR chain set to address (https://github.com/mobilecoinfoundation/mobilecoin/issues/2323)
This specifically addresses: https://github.com/mobilecoinfoundation/mobilecoin/issues/2325

### Future Work
Not addressing enclave issues in this PR
Migrating 
https://github.com/mobilecoinfoundation/mobilecoin/issues/2324

[Soundtrack of this PR](https://www.youtube.com/watch?v=DFu8P7teOt8)

### CODE REVIEW

File: transaction/core/test-utils/src/mint.rs [hunk 1]
```diff
@@ -23,16 +23,31 @@ use mc_util_from_random::FromRandom;
...
+/// keys that are allowed to sign it.
+///
+/// # Arguments
+/// `token_id` - The token id to use.
+/// `rng` - Randomness source.
+/// `nonce` - Nonce to use
+pub fn create_mint_config_tx_and_signers_for_set_nonce(
+    token_id: TokenId,
+    rng: &mut (impl RngCore + CryptoRng),
+    nonce: Vec<u8>,
```
Comment (iamalwaysuncomfortable) [inline transaction/core/test-utils/src/mint.rs:43]: Does this require a separate function that's part of the public API? Feels like it could be an Option parameter on the original function.
